app/models/grda_warehouse/group_viewable_entity.rb

```diff
@@ -21,5 +23,84 @@ class GroupViewableEntity < GrdaWarehouseBase
     scope :viewable_by, ->(user) do
       where(access_group_id: user.access_groups.pluck(:id))
     end
+
+    def entity_name
+      collection_name = collection&.name || 'Unknown Collection'
+      entity_display_name = case entity_type
+      when 'GrdaWarehouse::Lookups::CocCode'
+        "CoC Code: #{entity&.coc_code || entity_id}"
+      when 'GrdaWarehouse::Hud::Project'
+        "Project: #{entity&.ProjectName || entity_id}"
+      when 'GrdaWarehouse::Hud::Organization'
+        "Organization: #{entity&.OrganizationName || entity_id}"
+      when 'GrdaWarehouse::DataSource'
+        "Data Source: #{entity&.name || entity_id}"
+      when 'GrdaWarehouse::ProjectAccessGroup'
+        "Project Group: #{entity&.name || entity_id}"
```

Comment:
I don't know if this language is correct, but I believe this version of project groups is used to govern access to projects while the other governs access to project groups.
```suggestion
        "Project Group for Project Access: #{entity&.name || entity_id}"
```